index.js

```diff
@@ -786,7 +786,10 @@
 
   //  Array$prototype$extend :: Array a ~> (Array a -> b) -> Array b
   function Array$prototype$extend(f) {
-    return [f(this)];
+    var xs = this;
+    return xs.map(function(_, i) {
+      return f(xs.slice(i));
+    });
```

Comment:
I find this approach slightly nicer, as it obviates the need for a variable to keep track of `this`:

```javascript
return this.map(function(_, idx, xs) { return f(xs.slice(idx)); });
```

_Note the use of `idx` rather than `i` to match existing code._